android/src/main/java/com/reactnativedidomi/DidomiModule.kt

```diff
@@ -968,15 +976,46 @@ class DidomiModule(reactContext: ReactApplicationContext) : ReactContextBaseJava
         promise.resolve(transaction.commit())
     }
 
+    @ReactMethod
+    fun syncAcknowledged(callbackIndex: Int, promise: Promise) {
+        val result = syncAcknowledgedCallbacks[callbackIndex]?.let { it() }
+        syncAcknowledgedCallbacks.remove(callbackIndex)
+        if (result != null) {
+            promise.resolve(result)
+        } else  {
+            promise.reject(java.lang.IllegalStateException("SyncAcknowledged: Native callback not found. The method can be called only once."))
```

Comment:
I'd suggest adding something like this:
```suggestion
            promise.reject(java.lang.IllegalStateException("SyncAcknowledged: Native callback not found. This callback can be called only once and it has already been called."))
```

Reply:
Actually we can only say that the callback is not present. With the current implementation it is almost certain that it's because the method was already called, but in theory there could be other reasons.